metadata-service/openapi-entity-servlet/src/main/java/io/datahubproject/openapi/v2/delegates/EntityApiDelegateImpl.java

```diff
@@ -992,8 +1056,38 @@ public ResponseEntity<BusinessAttributeInfoAspectResponseV2> getBusinessAttribut
   }
 
   public ResponseEntity<Void> headBusinessAttributeInfo(String urn) {
+    if (checkBusinessAttributeFlagFromUrn(urn)) {
+      throw new UnsupportedOperationException(BUSINESS_ATTRIBUTE_ERROR_MESSAGE);
+    }
     String methodName =
         walker.walk(frames -> frames.findFirst().map(StackWalker.StackFrame::getMethodName)).get();
     return headAspect(urn, methodNameToAspectName(methodName));
   }
+
+  private boolean checkBusinessAttributeFlagFromUrn(String urn) {
+    try{
+      if (checkBusinessAttributeFlagFromEntityType(Urn.createFromString(urn).getEntityType())) {
+        return true;
+      }
+    } catch (URISyntaxException e) {
+      log.error(String.format("Can't convert urn = %s, Error = %s", urn, e.getMessage()));
+      return true;
+    }
+    return false;
+  }
+
+  private boolean checkBusinessAttributeFlagFromEntityType(String entityType) {
+    log.info("System getenv ba : {} ",System.getenv("BUSINESS_ATTRIBUTE_ENTITY_ENABLED"));
+    if (entityType.equals("businessAttribute") && !getSystemEnvForBusinessAttribute()) {
+      log.error(BUSINESS_ATTRIBUTE_ERROR_MESSAGE);
+      return true;
+    }
+    return false;
+  }
+
+  private boolean getSystemEnvForBusinessAttribute() {
+    return System.getenv("BUSINESS_ATTRIBUTE_ENTITY_ENABLED") == null ? false :
```

Comment:
convert this expression to --> System.getenv("BUSINESS_ATTRIBUTE_ENTITY_ENABLED")!=null && Boolean.parseBoolean(System.getenv("BUSINESS_ATTRIBUTE_ENTITY_ENABLED"))